subsys/bluetooth/controller/hci/hci.c

```diff
@@ -1542,74 +1547,79 @@ static void le_setup_iso_path(struct net_buf *buf, struct net_buf **evt)
 
 	rp = hci_cmd_complete(evt, sizeof(*rp));
 	rp->status = status;
-	rp->handle = cmd->handle;
```

Comment:
No reconversion going on. Why not leave it at cmd->handle? Is there a risk of cmd->.. being overwritten?

Reply:
refer to https://github.com/zephyrproject-rtos/zephyr/blob/master/subsys/bluetooth/host/hci_core.c#L9243

Its been longtime... things may have changed... but I believe, the command buffer is reused to send back command complete

Reply:
sorry, I am trying to make different commit, for reducing the conversion, fixing the overwriting etc... lately, merged contribution have this inconsistency of overwrites and missing conversions.... keeping this PR in draft till I get the different commits correct

Reply:
practically the overwrites may not be noticed, but the correct needs to be correct so that in the future other parameters returns should not get overwritten.

Reply:
I think fixing the overwrites is way more important than eliminating some endianness conversions (which will be NOPs for most architectures anyway). I suspect that a reasonably smart compiler will eliminate the conversion/re-conversion, but I need to check whether this is actually the case.

Reply:
As I am developing on little-endian machines, I did not see any conversion code in my builds before and after the change in this PR (flash usage remained same, as expected). Please check in big-endian machine, if the PR does not increase code size compared to before the PR.

Reply:
I quick check did not reveal a code size difference between the two patterns below. I think the first pattern is more readable, but I will not object if other reviewers find the second pattern preferrable.
```
handle = sys_le16_to_cpu(cmd->handle)
...
rp->handle = sys_cpu_to_le16(handle)
```

```
handle_le16 = cmd->handle;
handle = sys_le16_to_cpu(handle_le16)
...
rp->handle = handle_le16
```

Reply:
I have done an experiment to use `sys_be16_to_cpu` on a little-endian machine and compared the assembly instructions:

![image](https://user-images.githubusercontent.com/6350656/100569967-df3f0380-32cf-11eb-8b91-90bae67e2008.png)

![image](https://user-images.githubusercontent.com/6350656/100569780-5d4eda80-32cf-11eb-82b8-4980e0d3d07e.png)

There is clearly added instructions on re-conversions in the first pattern compared to the use of auto-variable (register r4) to remember the value before conversion. (I will assume this could be observed when using `sys_le16_to_cpu` on a big-endian machine. And of course, this could be compiler dependent when optimizations is used).